File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/LowestBackPriceFunction.java
package cz.fb.manaus.reactor.profit.progress.function;

import cz.fb.manaus.core.model.Side;
import org.springframework.stereotype.Component;

@Component
public class LowestBackPriceFunction extends AbstractPriceReduceFunction {

    public LowestBackPriceFunction() {
        super(Side.BACK, Math::min);
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/LowestBackPriceFunction.kt
package cz.fb.manaus.reactor.profit.progress.function

import cz.fb.manaus.core.model.Side
import org.springframework.stereotype.Component
import kotlin.math.min

@Component
object LowestBackPriceFunction : ProgressFunction by PriceReduceFunction(Side.BACK, { a, b -> min(a, b) })
